src/kernel/time/clock_tick.c

```diff
@@ -27,15 +25,12 @@ void clock_tick_handler(int irq_num, void *dev_id) {
 	struct clock_source *cs = (struct clock_source *) dev_id;
 
 	assert(cs);
-	if (++cs->jiffies_cnt == (1 << SLOWDOWN_SHIFT)) {
-		cs->jiffies_cnt = 0;
-		cs->jiffies++;
-
-		if (cs_jiffies->event_device &&	irq_num == cs_jiffies->event_device->irq_nr) {
-			if (timer_strat_need_sched(cs_jiffies->jiffies)) {
-				lthread_launch(&clock_handler_lt);
-			}
-		}
+
+	cs->jiffies_cnt = 0;
+	cs->jiffies++;
+
+	if (timer_strat_need_sched(cs->jiffies)) {
```

Comment:
Also, I proposed here to change `cs` -> `cs_jiffies` to make this code looks more clear.

Reply:
Because we are using cs_jiffies in lthread below.

Reply:
Or alternatevely you can set `assert (cs == cs_jiffies)` instead.

Reply:
agree. done